video/image_writer.c

```diff
@@ -128,6 +129,24 @@ static bool write_lavc(struct image_writer_ctx *ctx, mp_image_t *image, FILE *fp
     avctx->height = image->h;
     avctx->color_range = mp_csp_levels_to_avcol_range(image->params.color.levels);
     avctx->pix_fmt = imgfmt2pixfmt(image->imgfmt);
+
+    /*
+     * tagging avctx->bits_per_raw_sample indicates the number of significant
+     * bits. For example, if the original video was 10-bit, and the GPU buffer is
+     * 16-bit, this tells lavc that only 10 bits are significant. lavc encoders may
+     * ignore this value, but some codecs can make use of it (for example, PNG's
+     * sBIT chunk or JXL's bit depth header)
+     */
+    if (ctx->opts->high_bit_depth && ctx->original_format.id != image->imgfmt) {
```

Comment:
why only if the pixfmt is different?

Reply:
If the pixel format is the same, then there's no need for this metadata as it would set it to the max of the pixel format.

Reply:
That's fine it just tripped me up.

Reply:
One thing to note is that `ctx->opts->high_bit_depth` is no guarantee that you get a bit depth > 8, so you may need to adjust the check.

> because bits_per_raw_sample shouldn't be set for 8-bit screenshots.

btw, why is that?

Reply:
The idea was that higher bit depth video can be tagged appropriately (e.g. 10bit) although I suppose theoretically you could have a lower bit depth PNG.

Reply:
Ah, I thought there was a technical reason why that must not happen not just "it's pointless".
re the problem I mentioned: drop the `ctx->opts->high_bit_depth` check then?